deps/rabbit/src/amqqueue.erl

```diff
@@ -120,6 +122,17 @@
           type_state = #{} :: map() | ets:match_pattern()
          }).
 
+%% A subset of the amqqueue record to avoid looking up the full amqqueue record
+%% when delivering a message to a target queue.
+-record(queue_target,
```

Comment:
As we only ever read the full record would it be better bundle `type`, `pid` and `extra_bcc` into a single ETS field and use ets:lookup_elemnt to avoid copying the queue name back again? `name` is a resource so contains both the queue name and vhost which we already know from routing. If vhost and qname are both 64 bytes we end up copying the full binary back to the queue process.

Reply:
That's a great idea. I pushed a commit fixing this.